### PR TITLE
feat(validators): host, ipv4 and domain name validators

### PR DESCRIPTION
**Pull request by priyajeet**
@hapi/address is used by @hapi/Joi JSON schema validator, adds about 7kb

BREAKING CHANGE: Adds peer dependency of @hapi/address used for email and domain validation

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @priyajeet has signed the CLA. Thanks for the pull request!

File: src/utils/validators.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+function ipv4AddressValidator(value: string): boolean {
+    // @see https://www.oreilly.com/library/view/regular-expressions-cookbook/9780596802837/ch07s16.html
+    const regex = /^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/;
+    return regex.test(value);
+}
+
+function domainNameValidator(value: string): boolean {
+    return Address.domain.isValid(value);
```
Comment (jstoffan) [inline src/utils/validators.js]: Does this need to be in Elements or should only apps that need such a validator include the dependency?

**Comment by jstoffan** [comment]
Th PR description has the BREAKING CHANGE modifier, but the commit message description currently doesn't.

File: src/utils/validators.js (hunk 1 above)
Comment (priyajeet) [reply src/utils/validators.js]: It is now used in UIE

File: src/features/unified-share-modal/EmailForm.js [hunk 2]
```diff
@@ -195,9 +196,7 @@ class EmailForm extends React.Component<Props, State> {
     };
 
     isValidEmail = (text: string): boolean => {
-        // TODO-AH: make this a constant somewhere
-        const emailValidation = /^[^\s<>@,]+@[^\s<>@,/\\]+\.[^\s<>@,]+$/i;
-        return emailValidation.test(text);
+        return emailValidator(text);
```
Comment (priyajeet) [inline src/features/unified-share-modal/EmailForm.js]: @booc0mtaco

**Pull request merged**